internal/pipe/ko/ko.go

```diff
@@ -0,0 +1,301 @@
+// Package ko implements the pipe interface with the intent of
+// building OCI compliant images with ko.
+package ko
+
+import (
+	stdcontext "context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"github.com/goreleaser/goreleaser/internal/semerrgroup"
+	"golang.org/x/tools/go/packages"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/google/ko/pkg/build"
+	"github.com/google/ko/pkg/commands/options"
+	"github.com/google/ko/pkg/publish"
+	"github.com/goreleaser/goreleaser/internal/ids"
+	"github.com/goreleaser/goreleaser/internal/tmpl"
+	"github.com/goreleaser/goreleaser/pkg/config"
+	"github.com/goreleaser/goreleaser/pkg/context"
+)
+
+// Pipe that build OCI compliant images with ko.
+type Pipe struct{}
+
+func (Pipe) String() string { return "ko" }
+func (Pipe) Skip(ctx *context.Context) bool {
+	return ctx.SkipKo || len(ctx.Config.Kos) == 0
+}
+
+// Default sets the Pipes defaults.
+func (Pipe) Default(ctx *context.Context) error {
+	ids := ids.New("kos")
+	for _, ko := range ctx.Config.Kos {
+		if err := setConfigDefaults(&ko); err != nil {
+			return err
+		}
+		ids.Inc(ko.ID)
+	}
+	return ids.Validate()
+}
+
+func setConfigDefaults(cfg *config.Ko) error {
+	cfg.Push = true
+
+	if cfg.ID == "" {
+		cfg.ID = "default"
+	}
+
+	if cfg.BaseImage == "" {
+		cfg.BaseImage = "cgr.dev/chainguard/static" // TODO: we can discuss on this
```

Comment:
I'd love to have this come from a const in `ko`, rather than hard-code it again here. If `ko` changes its default base again, we can have goreleaser update easily as well just by updating the ko dependency.

Reply:
https://github.com/ko-build/ko/pull/904